S3/Utils.py

```diff
@@ -244,27 +266,29 @@ def check_bucket_name(bucket, dns_strict=True):
         if invalid:
             raise S3.Exceptions.ParameterError("Bucket name '%s' contains disallowed character '%s'. The only supported ones are: us-ascii letters (a-z, A-Z), digits (0-9), dot (.), hyphen (-) and underscore (_)." % (bucket, invalid.groups()[0]))
 
+    # The above block pre-filters some things.   But the lower stuff has to
+    # be more permissive to allow for what's allowed in the least restrictive
+    # filters above.
+
     if len(bucket) < 3:
         raise S3.Exceptions.ParameterError("Bucket name '%s' is too short (min 3 characters)" % bucket)
-    if len(bucket) > 255:
-        raise S3.Exceptions.ParameterError("Bucket name '%s' is too long (max 255 characters)" % bucket)
-    if dns_strict:
-        if len(bucket) > 63:
-            raise S3.Exceptions.ParameterError("Bucket name '%s' is too long (max 63 characters)" % bucket)
-        if re.search("-\.", bucket, re.UNICODE):
-            raise S3.Exceptions.ParameterError("Bucket name '%s' must not contain sequence '-.' for DNS compatibility" % bucket)
-        if re.search("\.\.", bucket, re.UNICODE):
-            raise S3.Exceptions.ParameterError("Bucket name '%s' must not contain sequence '..' for DNS compatibility" % bucket)
-        if not re.search("^[0-9a-z]", bucket, re.UNICODE):
-            raise S3.Exceptions.ParameterError("Bucket name '%s' must start with a letter or a digit" % bucket)
-        if not re.search("[0-9a-z]$", bucket, re.UNICODE):
-            raise S3.Exceptions.ParameterError("Bucket name '%s' must end with a letter or a digit" % bucket)
+    if len(bucket) > max_length:
+        raise S3.Exceptions.ParameterError("Bucket name '%s' is too long (max %d characters)" % (bucket, max_length))
+    if re.search("-\.", bucket, re.UNICODE):
+        raise S3.Exceptions.ParameterError("Bucket name '%s' must not contain sequence '-.' for DNS compatibility" % bucket)
+    if re.search("\.\.", bucket, re.UNICODE):
```

Comment:
Same a previous one, you could probably simplify that in:
`if '..' in bucket`

And maybe merge the 2 tests in fact!